# Secondary Dataset: Website Phishing

**Source:** UCI Machine Learning Repository  
https://archive.ics.uci.edu/dataset/379/website+phishing

This notebook loads and explores the Website Phishing dataset for our CS 418 phishing research project.

In [5]:
import arff
import polars as pl

In [6]:
with open("../datasets/website-phishing/PhishingData.arff", "r") as file:
    dataset = arff.load(file)

In [7]:
dataset.keys()

dict_keys(['description', 'relation', 'attributes', 'data'])

In [8]:
# create the Polars dataframe
column_names = [attribute[0] for attribute in dataset["attributes"]]

df = pl.DataFrame(
    dataset["data"],
    schema=column_names,
    orient="row"
)

df.head()

SFH,popUpWidnow,SSLfinal_State,Request_URL,URL_of_Anchor,web_traffic,URL_Length,age_of_domain,having_IP_Address,Result
str,str,str,str,str,str,str,str,str,str
"""1""","""-1""","""1""","""-1""","""-1""","""1""","""1""","""1""","""0""","""0"""
"""-1""","""-1""","""-1""","""-1""","""-1""","""0""","""1""","""1""","""1""","""1"""
"""1""","""-1""","""0""","""0""","""-1""","""0""","""-1""","""1""","""0""","""1"""
"""1""","""0""","""1""","""-1""","""-1""","""0""","""1""","""1""","""0""","""0"""
"""-1""","""-1""","""1""","""-1""","""0""","""0""","""-1""","""1""","""0""","""1"""


In [9]:
df.shape

(1353, 10)

In [10]:
df.schema

Schema([('SFH', String),
        ('popUpWidnow', String),
        ('SSLfinal_State', String),
        ('Request_URL', String),
        ('URL_of_Anchor', String),
        ('web_traffic', String),
        ('URL_Length', String),
        ('age_of_domain', String),
        ('having_IP_Address', String),
        ('Result', String)])

In [11]:
df["Result"].value_counts()

Result,count
str,u32
"""-1""",702
"""0""",103
"""1""",548


In [12]:
df.null_count()

SFH,popUpWidnow,SSLfinal_State,Request_URL,URL_of_Anchor,web_traffic,URL_Length,age_of_domain,having_IP_Address,Result
u32,u32,u32,u32,u32,u32,u32,u32,u32,u32
0,0,0,0,0,0,0,0,0,0


## Dataset Summary

- **Rows:** 1,353
- **Columns:** 10
- **One row represents:** one website with phishing-related characteristics and a classification result
- **Missing values:** none
- **Column types:** all columns are loaded as strings
- **Target column:** `Result`
  - `-1` = phishing
  - `0` = suspicious
  - `1` = legitimate
- **Class distribution:**
  - 702 phishing websites
  - 103 suspicious websites
  - 548 legitimate websites

### Columns of Interest

- `URL_Length`
- `having_IP_Address`
- `SSLfinal_State`
- `age_of_domain`
- `web_traffic`
- `Request_URL`
- `URL_of_Anchor`
- `Result`